contents/tutorials/analyze-surveys-with-chatgpt.md

```diff
@@ -0,0 +1,385 @@
+
+Then, install [OpenAI's Node library ](https://github.com/openai/openai-node) and initialize it in the code. 
+
+```bash
+npm install openai
+```
+
+```js
+// in survey-analyzer.js
+
+const OpenAI = require("openai");
+
+const openai = new OpenAI({
+    apiKey: "<your-api-key>"
+});
+
+// rest of the code
+```
+
+Next, for each survey answer, we'll make an API call to [Chat completions API](https://platform.openai.com/docs/guides/gpt/chat-completions-api) to extract the following information:
+
+1. The sentiment of the answer i.e., is it positive or negative?
+2. The theme of the answer.
+
+To do so, the API method requires 2 parameters:
+
+- **[model](https://platform.openai.com/docs/models/overview)** – which ChatGPT model to use. We use `gpt-3.5-turbo-16k` (we explain why later in this post).
+- [messages](https://platform.openai.com/docs/guides/gpt/chat-completions-api) – i.e., the task you are requesting ChatGPT to perform. You can find more details on the required format in the [API docs](https://platform.openai.com/docs/guides/gpt/chat-completions-api).
+
+With this in mind, here's what our code looks like:
+
+```js
+
+// rest of the code
+
+const analyzeSurveyAnswers = async () => {
+  const surveyAnswers = await csvToArr('answers.csv')
+  for (const surveyAnswer of surveyAnswers) {
+    try {
+      const result = await openai.chat.completions.create({
+        model: "gpt-3.5-turbo-16k",
+        messages: [
+          {
+            "role": "system", 
+            "content": `Your job is to analyze survey answers. For each answer, extract the sentiment (positive or negative only) as well as the theme in the answer. 
+
+              You must only respond with a JSON. This JSON must contain the keys "survey_answer", "sentiment", and "theme". 
+              
+              Here's an example of an valid answer: 
+              \`\`\`
+              {
+                "survey_answer": "The product has potential, but it's not very intuitive to use. Please simplify the interface",
+                "sentiment": "negative",
+                "theme": "user interface"
+              }
+              \`\`\`
+              `
+          },
+          {
+            "role": "user", 
+            "content": `Analyze this survey answer: ${JSON.stringify(surveyAnswer.answer)}`
+          }
+        ],
+      });
+
+      const analyzedAnswer = JSON.parse(result.choices[0].message.content);
+    } catch (error) {
+      console.log(`An error occurred: ${error}`)
+    }
+  }
+}
+```
+
+Finally, we'll save the answer from ChatGPT into a CSV file. This way we'll be able to analyze the results later:
+
+```js
+const analyzeSurveyAnswers = async () => {
+    // rest of function code...
+
+      const analyzedAnswer = JSON.parse(result.choices[0].message.content);
```

Comment:
Another (potentially dumb, feel free to ignore) idea, capturing these analyzed answers as a PostHog event? Creating some insights based on them in PostHog?

Reply:
Then you could have percentage of positive versus negative, common themes as insights